carbon-design-system/carbon-addons-iot-react
# fix(table): removed needing hasRowExpansion when using hasRowNesting

Closes #1463

**Summary**

- Removed the need to have hasRowExpansion when only needing hasRowNesting

**Change List (commits, features, bugs, etc)**

- TableHead now gets `hasRowNesting` and adds a column if present OR `hasRowExpansion` is present
- Updated documentation in table story

**Acceptance Test (how to verify the PR)**

- Verify table head produces an extra column when only hasRowNesting is enabled
- Go to https://deploy-preview-1842--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table--simple-stateful-example and set `hasRowSelection` to single and verify the header renders correctly when only `hasRowNesting` or `hasRowExpansion` is true.
